### PR TITLE
[Elasticsearch-xpack] [Metricbeat] Fix elasticsearch-xpack error when Machine Learning is explicitly disabled

### LINKED ISSUE #28087: Metricbeat monitoring module elasticsearch-xpack error when Machine Learning is explicitly disabled
## Summary
When using the `elasticsearch-xpack` module to monitor an Elasticsearch node, this one fails if Machine Learning is explicitly disabled (`xpack.ml.enabled: false`).

```
Error fetching data for metricset elasticsearch.ml_job: HTTP error 400 in : 400 Bad Request
```

This is due to the response from `GET _ml/anomaly_detectors/_all/_stats` and is independent of license level.

If Machine Learning is not disabled, the response for this API call will be at least:
```
{
  "count": 0,
  "jobs": []
}
```

When setting `xpack.ml.enabled: false`, the response changes to:
```
{
  "error": "no handler found for uri [/_ml/anomaly_detectors/_all/_stats] and method [GET]"
}
```

Which won't be understood by the module as "I disabled ML".

For license levels where ML is not a feature, this setting has no impact, but for licenses that could have ML, one might want to explicitly disable it and this will cause monitoring errors.

- Steps to Reproduce: 
1. Install a single node Elasticsearch cluster 
2. Set `xpack.ml.enabled: false` in the `elasticsearch.yml` file
3. Issue `GET _ml/anomaly_detectors/_all/_stats`
```
{
  "error": "no handler found for uri [/_ml/anomaly_detectors/_all/_stats] and method [GET]"
}
```

## Technical details 

I investigated the issue and one way to fix it is to check if the ML is enabled to avoid getting the error shown. We can use the [API](https://www.elastic.co/guide/en/elasticsearch/reference/current/info-api.html) to get the information - under `features` we can check if `ml` is `enabled/available` upfront.

### PR DESCRIPTION
**Pull request by jennypavlova**
Closes #28087 

- Bug

## Proposed commit message

This PR fixes `elasticsearch-xpack` error when Machine Learning is explicitly disabled by checking if the Machine Learning is enabled

### CODE REVIEW

File: metricbeat/module/elasticsearch/ml_job/ml_job.go [hunk 1]
```diff
@@ -53,12 +55,19 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right format
 func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 	shouldSkip, err := m.ShouldSkipFetch()
+	mLAvailability, mlErr := m.checkMLAvailability()
```
Comment (klacabane) [inline metricbeat/module/elasticsearch/ml_job/ml_job.go:58]: we can move this after line 64 so we don't make the request if not necessary

Comment (jennypavlova) [reply metricbeat/module/elasticsearch/ml_job/ml_job.go]: Moved, thanks 👍

File: metricbeat/module/elasticsearch/ml_job/ml_job.go [hunk 2]
```diff
@@ -72,3 +81,17 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 
 	return eventsMapping(r, info, content, m.XPackEnabled)
 }
+
+func (m *MetricSet) checkMLAvailability() (message string, err error) {
```
[the remark this answers is not in the document]
Comment (jennypavlova) [reply metricbeat/module/elasticsearch/ml_job/ml_job.go]: Sure, changed ✅

File: metricbeat/module/elasticsearch/ml_job/ml_job_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ml_job
+
+import (
+	"io/ioutil"
```
Comment (jennypavlova) [inline metricbeat/module/elasticsearch/ml_job/ml_job_test.go:21]: I am not sure why I see these errors - the same import exists in other files 🤔

**Comment by jennypavlova** [comment]
@klacabane Thank you for the review! 
I fixed the comments and the test and don't see the error when `xpack.ml.enabled` is set to `false`. 
I would need some help with the linting errors (I tried some formatters and tips I found but nothing worked so far) and testing with the actual ML job running locally (the case when the ML is enabled in elasticsearch - I am not sure what the expected log is in this case and if any additional configuration is needed).

**Review by klacabane — APPROVED** [review]

**Pull request merged**